Installing compiler - incompatible with GCC-7.x

<!--- Provide a general summary of the issue in the Title above -->

## Current Behavior
When installing the compiler from development on debian based systems, no apparent faults are noticed in output.
However, when compiling any encore file the code generation breaks giving the subsequent error:

```
alexis@alexis-zen:~/encoretest$ encorec plusplus.enc -c
/home/alexis/github/encore/release/lib/libponyrt.a(pool.o): In function `pool_pull':
/home/alexis/github/encore/src/runtime/pony/libponyrt/mem/pool.c:763: undefined reference to `__atomic_compare_exchange_16'
/home/alexis/github/encore/src/runtime/pony/libponyrt/mem/pool.c:763: undefined reference to `__atomic_compare_exchange_16'
/home/alexis/github/encore/release/lib/libponyrt.a(pool.o): In function `pool_push':
/home/alexis/github/encore/src/runtime/pony/libponyrt/mem/pool.c:726: undefined reference to `__atomic_compare_exchange_16'
/home/alexis/github/encore/release/lib/libponyrt.a(pool.o): In function `pool_pull':
/home/alexis/github/encore/src/runtime/pony/libponyrt/mem/pool.c:763: undefined reference to `__atomic_compare_exchange_16'
/home/alexis/github/encore/release/lib/libponyrt.a(pool.o): In function `pool_push':
/home/alexis/github/encore/src/runtime/pony/libponyrt/mem/pool.c:726: undefined reference to `__atomic_compare_exchange_16'
/home/alexis/github/encore/release/lib/libponyrt.a(pool.o):/home/alexis/github/encore/src/runtime/pony/libponyrt/mem/pool.c:726: more undefined references to `__atomic_compare_exchange_16' follow
clang: error: linker command failed with exit code 1 (use -v to see invocation)
 *** Compilation failed with exit code 1 ***
```

## Temporary Workaround
*This is a hack and should not be the correct solution*
1. uninstall gcc
2. install gcc-6
3. make a symlink of /usr/bin/gcc to /usr/bin/gcc-6
4. cd PATH-TO-ENCORE && make -B

I did these steps after running debian-install.sh first,to ensure all other instructions had gone through before By uninstalling gcc you also delete local versions of cabal-install, ghc and other possible programs.


## Possible Solution

When compiling ponyrt, make sure to link with `libatomic` and possibly when compiling encore-files, make sure to include `-latomic`

This issue have been managed in the pony-compiler with following PR https://github.com/ponylang/ponyc/pull/2134
although this is for ponyc, not ponyrt

--- a/src/runtime/pony/premake4.lua
+++ b/src/runtime/pony/premake4.lua
@@ -25,6 +25,11 @@ if premake.override then
   end)
 end
 
+if os.execute("clang -v") == 0 then
+  premake.gcc.cc  = 'clang'
+  premake.gcc.cxx = 'clang++'
+end
+
 function use_flto()
   buildoptions {
     "-O3",

